credit/physics_core.py

```diff
@@ -12,10 +12,77 @@
 """
 
 import torch
-from typing import Dict
-from credit.physics_constants import RAD_EARTH, GRAVITY
+import torch.nn as nn
+from typing import Dict, Any, Optional
+from credit.physics_constants import RAD_EARTH, RVGAS, RDGAS, EPSGAS, GRAVITY, RHO_WATER, LH_WATER
 
 
+def compute_density(pressure, temperature, specific_humidity):
+    """
+    compute density given pressure (Pa), temperature (K), and specific humidity (kg/kg)
+    """
+    virtual_temperature = compute_virtual_temperature(temperature, specific_humidity)
+    density = pressure / (RDGAS * virtual_temperature)
+    return density
+
+def compute_virtual_temperature(temperature, specific_humidity):
+    """ref: metpy"""
+    mixing_ratio = specific_humidity / (1 - specific_humidity)
+    temperature_virtual = temperature * (mixing_ratio + EPSGAS) / (EPSGAS * (1 + mixing_ratio))
+    return temperature_virtual
+
+
+class compute_pressure_on_mlevs(nn.Module):
```

Comment:
Python class names should use camel case by convention with the first letter capitalized. Also, classes usually have a noun name rather than a verb. A preferred name would be `ModelLevelPressures`.